pastis/plotting.py

```diff
@@ -1000,57 +1000,58 @@ def natural_keys(text):
+    Creates surface deformation maps (not WFE) for localized wavefront aberrations.
+
+    The input mode coefficients 'mus' are in units of *WFE* and need to be grouped by segment, meaning the array holds
+    the mode coefficients as:
+        mode1 on seg1, mode2 on seg1, ..., mode'nmodes' on seg1, mode1 on seg2, mode2 on seg2 and so on.
 
     Parameters:
-    tel : class instance of the simulator for "instrument"
+    -----------
+    tel : class instance of internal simulator
+        the simulator to plot the surface maps for
     mus : 1d array
-        an array of segment requirements for all segment-level localized aberration modes, in units of nm.
-        each element in this array is a tolerance value per segment per mode
-        len(mus) = total number of modes * total number of segments
+        1d array of standard deviations for all modes on each segment, in nm WFE
     num_modes : int
-        the total number of local modes used to poke a segment.
-        For a harris segment mirror  it is 5 (taking only thermal basis map).
-        For a segmented mirror, it represents the total number of local zernike modes.
-    c_target : float
-        desired dark hole contrast for which the tolerancing is done
-    data_dir :  str
-        path to save the plot, if save=True
+        number of local modes used to poke each segment
+    mirror : str
+        'harris_seg_mirror' or 'seg_mirror', segmented mirror of simulator 'tel' to use for plotting
     cmin : float
-        minimum value for colorbar plot
+        minimum value for colorbar
     cmax : float
-        minimum value for colorbar plot
-    save : bool
-        whether to save the plot, default is False
+        maximum value for colorbar
+    data_dir : str, default None
+        path to save the plots; if None, then not saved to disk
+    fname : str, default None
+        file name for surface maps saved to disk
     """
-    nm_aber = CONFIG_PASTIS.getfloat('LUVOIR', 'calibration_aberration') * 1e-9
+    if fname is None:
+        fname = f'surface_on_{mirror}'
 
-    coeffs_mumaps = pastis.util.sort_1d_mus_per_actuator(mus, num_modes, tel.nseg)
+    mus_per_actuator = pastis.util.sort_1d_mus_per_actuator(mus, num_modes, tel.nseg)  # in nm
 
     mu_maps = []
     for mode in range(num_modes):
-        coeffs = coeffs_mumaps[mode] # in units of nm
-        if mirror == 'harris_sm':
-            tel.harris_sm.actuators = coeffs * nm_aber / 2 # in units of m
-            mu_maps.append(tel.harris_sm.surface)
-        if mirror == 'sm':
-            tel.sm.actuators = coeffs * nm_aber / 2
-            mu_maps.append(tel.sm.surface)  # in units of m, each nu_map is now of the order of 1e-9 m
-
-    for i in range(0, num_modes):
+        coeffs = mus_per_actuator[mode]
+        if mirror == 'harris_seg_mirror':
+            tel.harris_sm.actuators = coeffs * 1e-9 / 2  # in meters of surface
```

Comment:
1e-9 is a hardcoded parameter, I would still put nm_aber because this value used to create PASTIS matrix in first place. If we change nm_aber in config file for calculating a PASTIS matrix, I guess we have to adjust these values accordingly.

Reply:
> From equation 17 and 36 in [Laginja et al. (2021)](https://ui.adsabs.harvard.edu/abs/2021JATIS...75004L/abstract), units of coeffs here is directly scaled to nm_aber.

Exactly! This is why the only conversion needed is from nm to m when feeding the coefficients into the simulator 👍

Reply:
See also:
https://github.com/spacetelescope/PASTIS/pull/141#discussion_r1008935479